packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -96,91 +119,49 @@ export class SemanticViewerResultsRunner extends ResultsRunner {
-                return acc;
-            }, [] as VizIndexLayoutOptions[]),
-            valuesLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.METRIC) {
-                    acc.push({
-                        reference: column.reference,
-                    });
-                }
-                return acc;
-            }, [] as VizValuesLayoutOptions[]),
-            pivotLayoutOptions: this.columns.filter((column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                return field?.kind === FieldType.DIMENSION;
-            }),
+            indexLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
+            valuesLayoutOptions: this.metrics.map((metric) => ({
+                reference: metric.name,
+            })),
+            pivotLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
         };
     }
 
     defaultPivotChartLayout(): VizChartLayout | undefined {
-        const xColumn = this.columns.find((column) => {
-            const field =
-                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                    this.fields,
-                    column.reference,
-                );
-            return field?.kind === FieldType.DIMENSION;
-        });
+        const xColumn = this.dimensions[0];
+        const yColumn = this.metrics[0];
 
-        const yColumn = this.columns.find((column) => {
-            const field =
-                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                    this.fields,
-                    column.reference,
-                );
-            return field?.kind === FieldType.METRIC;
-        });
         if (!xColumn || !yColumn) {
             return;
         }
 
         return {
             x: {
-                reference: xColumn.reference,
-                type: this.getAxisType(xColumn),
+                reference: xColumn.name,
+                type: getVizIndexTypeFromDimensionType(xColumn.type),
             },
             y: [
                 {
-                    reference: yColumn.reference,
+                    reference: yColumn.name,
                 },
             ],
             groupBy: [],
         };
     }
 
-    static convertColumnsToVizColumns(
+    static convertColumnNamesToVizColumns(
```

Comment:
This is because the backend doesnt know types?

We should fix this on the backend